### PR TITLE
[AMSDK-11013] Support for sending update consent requests to Edge Network

### PR DESCRIPTION
**Pull request by emdobrin**
## Description

- Update EdgeHitProcessor to queue and process experience events and update consent events
- EdgeHit is now a protocol so it can be implemented for various types of requests
- In addition to the privacy optout check, ignore experience events when consent is set to no
- Sample app updates for consent events, requires future updates to use Consent public APIs
- Swift lint fixes in tests and disabled some rules

## How Has This Been Tested?

Unit and manual testing, functional tests should be covered in a separate PR

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [ ] I have signed the [Adobe Open Source CLA](https://opensource.adobe.com/cla.html).
- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: Sources/EdgeNetworkHandlers/EdgeHit.swift [hunk 1]
```diff
@@ -14,13 +14,83 @@ import AEPCore
...
+        self.consents = consents
+    }
+
+    func getType() -> ExperienceEdgeRequestType {
+        ExperienceEdgeRequestType.consent
+    }
+
+    func getPayload() -> String? {
+        let encoder = JSONEncoder()
+        encoder.outputFormatting = [.prettyPrinted]
```
[the remark this answers is not in the document]
Comment (emdobrin) [reply Sources/EdgeNetworkHandlers/EdgeHit.swift]: The request payload is also printed to debug logs.

**Review by PravinPK — APPROVED** [review]

File: Sources/EdgeNetworkHandlers/EdgeConsentUpdate.swift [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+//
+
+import AEPServices
+import Foundation
+
+/// A request for sending a consent update to the Adobe Experience Edge.
+/// An `EdgeConsentUpdate` is the top-level request object sent to Experience Edge to the set-consent endpoint.
+struct EdgeConsentUpdate: Encodable {
+    /// The IdentityMap at the moment of this request
+    let identityMap: IdentityMap?
```
Comment (kevinlind) [inline Sources/EdgeNetworkHandlers/EdgeConsentUpdate.swift:20]: nit: I think this is okay for now, in this PR, however when Edge reads the IdentityEdge XDM shared state, we should remove the IdentityMap class from Edge and use a `[String: Any]` type here.

**Review by kevinlind — APPROVED** [review]
Looks good. Added a comment, but doesn't require changes for this PR.

File: Sources/EdgeNetworkHandlers/EdgeConsentUpdate.swift (hunk 2 above)
Comment (emdobrin) [reply Sources/EdgeNetworkHandlers/EdgeConsentUpdate.swift:20]: 👍  should be handled by https://github.com/adobe/aepsdk-edge-ios/pull/165

**Pull request merged**